sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/FilePartitionReaderFactory.scala

```diff
@@ -16,20 +16,21 @@
  */
 package org.apache.spark.sql.execution.datasources.v2
 
-import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.{FileSourceOptions, InternalRow}
 import org.apache.spark.sql.connector.read.{InputPartition, PartitionReader, PartitionReaderFactory}
 import org.apache.spark.sql.errors.QueryExecutionErrors
 import org.apache.spark.sql.execution.datasources.{FilePartition, PartitionedFile}
 import org.apache.spark.sql.vectorized.ColumnarBatch
 
-abstract class FilePartitionReaderFactory extends PartitionReaderFactory {
+abstract class FilePartitionReaderFactory(options: FileSourceOptions)
```

Comment:
A simpler is to add it as a method
```
abstract class FilePartitionReaderFactory extends PartitionReaderFactory {
  protected def options: FileSourceOptions
  ...
}
```

Reply:
then we don't need to change its subclasses, as they are case class and already implicitly provided the `options` method.

Reply:
wow! thanks, that's simpler!